crates/client/rpc/src/lib.rs

```diff
@@ -214,6 +214,46 @@ where
 
         Ok(RPCContractClass::ContractClass(SierraContractClass::default()))
     }
+
+    /// Gets the status and details of a submitted transaction.
+    fn get_transaction_by_hash(
+        &self,
+        tx_hash: TransactionHash,
+    ) -> RpcResult<RpcTransaction> {
+
+        let tx_h256 = H256::from_str(&tx_hash).map_err(|e| {
+            error!("GetTransactionByHash: Failed to convert '{tx_hash}' to H256: {e}");
+            StarknetRpcApiError::InvalidHash
+        })?;
+
+        let substrate_block_hash = self
```

Comment:
Can you confirm that due to the subtrate architecture, this lines of codes are required to do a runtime call? Because several endpoints will not have the block as argument, and we can then make a thin wrapper to abstract this to them.